ui/pages/sample/hooks/useSamplePetnamesMetrics.ts

```diff
@@ -0,0 +1,102 @@
+import { useCallback, useContext } from 'react';
+import { Hex } from '@metamask/utils';
+import { MetaMetricsContext } from '../../../contexts/metametrics';
+import {
+  MetaMetricsEventCategory,
+  MetaMetricsEventName,
+} from '../../../../shared/constants/metametrics';
+
+/**
+ * Hook that provides tracking functions for the sample petnames feature
+ *
+ * @returns Object containing tracking functions for petnames actions
+ */
+export function useSamplePetnamesMetrics() {
+  const trackEvent = useContext(MetaMetricsContext);
+
+  /**
+   * Track when the petnames form is viewed
+   */
+  const trackPetnamesFormViewed = useCallback(() => {
```

Comment:
Do we have any opportunity to demonstrate a good `useMemo` example? Perhaps a cast where we return a non-primitive from a hook like an object or array?